crates/forge/bin/cmd/test/mod.rs

```diff
@@ -434,59 +383,60 @@ impl TestArgs {
                         TraceKind::Deployment => false,
                     };
 
+                    // Traces need to be identified for gas reports too.
+                    if should_include || self.gas_report {
+                        decoder.identify(arena, &mut local_identifier);
+                        decoder.identify(arena, &mut etherscan_identifier);
+                    }
+
                     if should_include {
                         decoded_traces.push(render_trace_arena(arena, &decoder).await?);
                     }
                 }
 
                 if !decoded_traces.is_empty() {
                     shell::println("Traces:")?;
-                    decoded_traces.into_iter().try_for_each(shell::println)?;
+                    for trace in &decoded_traces {
+                        shell::println(trace)?;
+                    }
                 }
 
                 if self.gas_report {
                     gas_report.analyze(&result.traces, &decoder).await;
                 }
 
-                // If the test failed, we want to stop processing the rest of the tests
-                if self.fail_fast && result.status == TestStatus::Failure {
-                    break 'outer
```

Comment:
This previously exited the loop without printing the summary. We also need to do this now since we push the results into the outcome at the end to avoid unnecessary cloning.